main.go

```diff
@@ -49,6 +50,24 @@ func gotest(args []string) int {
 
 	go consume(&wg, r)
 
+	sigc := make(chan os.Signal)
+	done := make(chan struct{})
+	defer func() {
+		done <- struct{}{}
+	}()
+	signal.Notify(sigc)
+
+	go func() {
+		for {
```

Comment:
You don't need the for loop, right? Just listening on sigc would be sufficient? It's gonna block and forward the signal if something happens. Am I wrong?

Reply:
My understanding is that this will forward multiple signals whereas omitting the for loop would only forward the first signal.

Reply:
yes, exactly. for example, i could send `SIGUSR1` and if the program doesn't react, `SIGKILL` later.

Reply:
Gotcha, thanks.